### PR TITLE
*: add keepalive

### PR DESCRIPTION
**Pull request by BusyJay**
So network failure can be detected. Tests also show that memory usage during partition can be reduced after this change.

### CODE REVIEW

**Comment by disksing** [comment]
LGTM.

**Comment by BusyJay** [comment]
@overvenus @siddontang PTAL

File: src/server/config.rs [hunk 1]
```diff
@@ -99,6 +101,10 @@ impl Default for Config {
             grpc_concurrent_stream: DEFAULT_GRPC_CONCURRENT_STREAM,
             grpc_raft_conn_num: DEFAULT_GRPC_RAFT_CONN_NUM,
             grpc_stream_initial_window_size: ReadableSize(DEFAULT_GRPC_STREAM_INITIAL_WINDOW_SIZE),
+            // There will be a heartbeat every secs, it's wired a connection will be idle for more
+            // than 10 senconds.
+            grpc_keepalive_time: ReadableDuration::secs(10),
+            grpc_keepalive_timeout: ReadableDuration::secs(3),
```
Comment (overvenus) [inline src/server/config.rs]: Maybe we need make sure that it is always longer than raft heartbeat.

**Comment by BusyJay** [comment]
PTAL

**Comment by overvenus** [comment]
> Tests also show that memory usage during partition can be reduced after this change.

Why does ping help TiKV reduce memory?

What will happen if keepalive timeout, in particular, ongonig calls?

**Comment by BusyJay** [comment]
> Why does ping help TiKV reduce memory?

Otherwise connection will hang so messages are pending in the channel, which will cause OOM in the end.

> What will happen if keepalive timeout, in particular, ongonig calls?

The connection will be dropped, ongoing calls will fail.

**Pull request merged**